agent/mibgroup/mibII/kernel_linux.c

```diff
@@ -438,7 +438,7 @@ linux_read_icmp_msg_stat(struct icmp_mib *icmpstat,
 
     if ((ret = linux_read_mibII_stats()) == -1) {
         return -1;
-    } else if (ret) {
+    } else if (!ret) {
```

Comment:
Please leave out the "else if (ret)" instead of modifying it.

Reply:
it's dead code, the condition is always false,  is it works as designed?

Reply:
Correct, the case where ret=1 was returned was removed in 142c4239771c6282d66a553d5692037c251a2f90

Reply:
> Correct, the case where ret=1 was returned was removed in [142c423](https://github.com/net-snmp/net-snmp/commit/142c4239771c6282d66a553d5692037c251a2f90)

maybe then delete this case?

Reply:
Correct, that is what Bart meant by "leave it out".

Reply:
sorry, apparently there was a misunderstanding, I thought just leave without changes